rust/operator-binary/src/command.rs

```diff
@@ -0,0 +1,165 @@
+    trino: &TrinoCluster,
+    user_authentication: Option<&TrinoAuthenticationConfig>,
+    s3_connection_spec: Option<&S3ConnectionSpec>,
+) -> Vec<String> {
+    let mut args = vec![
+        // copy config files to a writeable empty folder
+        format!(
+            "echo copying {conf} to {rw_conf}",
+            conf = CONFIG_DIR_NAME,
+            rw_conf = RW_CONFIG_DIR_NAME
+        ),
+        format!(
+            "cp -RL {conf}/* {rw_conf}",
+            conf = CONFIG_DIR_NAME,
+            rw_conf = RW_CONFIG_DIR_NAME
+        ),
+    ];
+
+    if let Some(auth) = user_authentication {
+        let user_data = auth.to_trino_user_data();
+        args.push(format!(
+            "echo '{data}' > {path}/{db}",
+            data = user_data,
+            path = USER_PASSWORD_DATA_DIR_NAME,
+            db = PASSWORD_DB
+        ));
+    }
+
+    // We need to read the provided s3 credentials from the secret operator / secret class folder
+    // and export it to the required env variables in order for trino to pick them up
+    // out of the config via e.g. ${ENV:S3_ACCESS_KEY}.
+    if let Some(S3ConnectionSpec {
+        credentials: Some(_),
+        ..
+    }) = s3_connection_spec
+    {
+        args.extend(vec![
+            format!(
+                "export {env_var}=$(cat {secret_dir}/{file_name})",
+                env_var = ENV_S3_ACCESS_KEY,
+                secret_dir = S3_SECRET_DIR_NAME,
+                file_name = SECRET_KEY_S3_ACCESS_KEY
+            ),
+            format!(
+                "export {env_var}=$(cat {secret_dir}/{file_name})",
+                env_var = ENV_S3_SECRET_KEY,
+                secret_dir = S3_SECRET_DIR_NAME,
+                file_name = SECRET_KEY_S3_SECRET_KEY
+            ),
+        ]);
+    }
+
+    // hive required?
+    if trino.spec.hive_config_map_name.is_some() {
+        args.extend(vec![
+            format!( "echo Writing HIVE connect string \"hive.metastore.uri=${{HIVE}}\" to {rw_conf}/catalog/{hive_properties}",
+                     rw_conf = RW_CONFIG_DIR_NAME, hive_properties = HIVE_PROPERTIES
+            ),
+            format!( "echo \"hive.metastore.uri=${{HIVE}}\" >> {rw_conf}/catalog/{hive_properties}",
+                     rw_conf = RW_CONFIG_DIR_NAME, hive_properties = HIVE_PROPERTIES
+            )])
+    }
+
+    // start command
+    args.push(format!(
+        "bin/launcher run --etc-dir={conf} --data-dir={data}",
+        conf = RW_CONFIG_DIR_NAME,
+        data = DATA_DIR_NAME
+    ));
+
+    vec![args.join(" && ")]
+}
+
+/// Generates the shell script to create key and truststores from the certificates provided
+/// by the secret operator.
+fn create_key_and_trust_store(directory: &str, alias_name: &str) -> Vec<String> {
+    vec![
+        format!("echo [{dir}] Creating truststore", dir = directory),
+        format!("keytool -importcert -file {dir}/ca.crt -keystore {dir}/truststore.p12 -storetype pkcs12 -noprompt -alias {alias} -storepass {password}",
```

Comment:
Please don't write into the secret-operator CSI directories. This currently works, but isn't guaranteed to keep working in the future.

Reply:
Yeah unfortunately we do this in nifi, zookeeper, kafka etc as well. Should we start adapting this here?

Reply:
Yes please.